src/client/datascience/notebookStorage/notebookModel.ts

```diff
@@ -23,6 +24,9 @@ export class NativeEditorNotebookModel extends BaseNotebookModel {
     public get cells(): ICell[] {
         return this._cells;
     }
+    public get kernelConnection(): KernelConnectionMetadata | undefined {
+        return this._kernelConnection;
```

Comment:
In NativeNotebooks, VSC keeps track of the current selected kernel (which contains the KernelConnectionMetadata).
I didn't want to create a property in the base INotebookModel to keep track of the KernelConnection as this is only required for interactive window & WebView notebooks. 

Thus for non-native notebooks, the KernelConnectionMetadta is stored against the model.
& for NativeNotebooks we let VSC store that information.

I could create a separate class that stores this information, but then its yet another class that needs to be added to interacgive window, native notebooks & webview notebooks, I think adding it into the NotebookModel (non-native notebook model) is simpler.

Also, anyone writing code for non-native notebooks will not be able to access this property (hence guarding against accidental usage in wrong places).